### ISSUE TITLE
[Crash] Getting crash with the creation of Exoplayer through builder method

### ISSUE THREAD

**Issue by raghav2945** [google/ExoPlayer#9956]
We started receiving crash over firebase for Exoplayer. And the error message is as below :

> Fatal Exception: java.lang.NullPointerException
> Attempt to invoke virtual method 'android.os.Looper com.google.android.exoplayer2.ExoPlayerImpl.getApplicationLooper()' on a null object reference

- ExoPlayer version number : 2.13.3
- Android version : 9/10/11/12 
- Android device : Samsumg/Google/LGE/Motorola

We just made some changes to the creation of Exoplayer because the construction with builder reported ANR. So we try to avoid this by using a kotlin coroutine, and the code looks like as below. However, we were suddenly start receiving crash instead of ANR. Although we are still receiving some ANR, it is not in same place.

```
CoroutineScope(Dispatchers.IO).launch {
            initExoPlayer()
        }

private fun initExoPlayer():
        SimpleExoPlayer = simpleExoPlayer
        ?: SimpleExoPlayer.Builder(applicationContext)
            .setAudioAttributes(
                AudioAttributes.Builder()
                    .setContentType(C.CONTENT_TYPE_MUSIC)
                    .setUsage(C.USAGE_MEDIA)
                    .build(),
                true
            )
            .setHandleAudioBecomingNoisy(true)
            .build()
            .also { simpleExoPlayer = it }

```
**Stack-trace for ANR before the changes**
```
main (native): tid=1 systid=3412
#00 pc 0x5e3e8 libc.so 
#01 pc 0x131bb3 libart.so 
#02 pc 0x4d409b libart.so 
#03 pc 0x4d3f21 libart.so 
       at android.media.AudioSystem.newAudioSessionId(AudioSystem.java)
       at android.media.AudioManager.generateAudioSessionId(AudioManager.java:2567)
       at com.google.android.exoplayer2.C.generateAudioSessionIdV21(C.java:1170)
       at com.google.android.exoplayer2.SimpleExoPlayer.<init>(SimpleExoPlayer.java:665)
       at com.google.android.exoplayer2.SimpleExoPlayer$Builder.build(SimpleExoPlayer.java:558)
       at com.audiomack.playback.SimpleExoPlayerImpl.init(SimpleExoPlayerImpl.java:24)
       at com.audiomack.MainApplication.onCreate(MainApplication.java:82)
       at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1192)
       at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7593)
       at android.app.ActivityThread.access$1500(ActivityThread.java:301)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2177)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:246)
       at android.app.ActivityThread.main(ActivityThread.java:8653)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
```

**Stack-trace for crash after the changes**
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'android.os.Looper com.google.android.exoplayer2.ExoPlayerImpl.getApplicationLooper()' on a null object reference
       at com.google.android.exoplayer2.SimpleExoPlayer.getApplicationLooper(SimpleExoPlayer.java:1200)
       at com.google.android.exoplayer2.SimpleExoPlayer.verifyApplicationThread(SimpleExoPlayer.java:1951)
       at com.google.android.exoplayer2.SimpleExoPlayer.getCurrentWindowIndex(SimpleExoPlayer.java:1652)
       at com.google.android.exoplayer2.analytics.AnalyticsCollector.generateEventTime(AnalyticsCollector.java:858)
       at com.google.android.exoplayer2.analytics.AnalyticsCollector.generateLoadingMediaPeriodEventTime(AnalyticsCollector.java:877)
       at com.google.android.exoplayer2.analytics.AnalyticsCollector.onBandwidthSample(AnalyticsCollector.java:722)
       at com.google.android.exoplayer2.upstream.BandwidthMeter$EventListener$EventDispatcher.lambda$bandwidthSample$0(BandwidthMeter.java:81)
       at com.google.android.exoplayer2.upstream.BandwidthMeter$EventListener$EventDispatcher$$ExternalSyntheticLambda0.run(BandwidthMeter.java:8)
       at android.os.Handler.handleCallback(Handler.java:790)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6543)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:440)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:810)
```

**Comment by ojw28** [comment]
It looks like `SimpleExoPlayer.getApplicationLooper` is being called before `player` has been set. @tonihei - Do your recent changes to merge logic into `ExoPlayerImpl` resolve this?

**Comment by tonihei** [comment]
This was fixed by https://github.com/google/ExoPlayer/commit/f31894616c02d0c947790c3228f48c1a28833854, first published in 2.14.0.
